src/approaches/pg3_approach.py

```diff
@@ -61,14 +62,18 @@ def _solve(self, task: Task, timeout: int) -> Callable[[State], Action]:
         policy."""
         skeleton = []
         atoms_sequence = []
+        current_objects = set()
         atoms = utils.abstract(task.init, self._initial_predicates)
         atoms_sequence.append(atoms)
+        for key in task.init.data:
```

Comment:
you can save a few lines and do:
`current_objects = set(task.init)`
to understand why this works, check out the `__iter__()` method of `State` in structs.py

Reply:
Oh thats pretty cool. Thanks!